content/300-guides/050-database/870-using-prisma-with-mongodb.mdx

```diff
@@ -0,0 +1,351 @@
+
+- **Defining IDs**: MongoDB documents have an `_id` field that contain an [ObjectID](https://www.mongodb.com/docs/manual/reference/bson-types/#std-label-objectid). This needs to be mapped to a Prisma field. For more information, see [Defining IDs in MongoDB](/concepts/components/prisma-schema/data-model#defining-ids-in-mongodb).
+
+- **Migrating old data to new schemas**: In relational databases, all your data must match your schema. If you change the type of a particular field in your schema when you migrate, all the data must also be updated to match. In contrast, MongoDB does not enforce any particular schema, so you need to take care when migrating. For more information, see [How to migrate old data to new schemas](#how-to-migrate-old-data-to-new-schemas).
+
+- **Introspection**: When you introspect an existing database, you will get a schema with no relations and will need to add the missing relations in manually. For more information, see [How to add in missing relations after Introspection](#how-to-add-in-missing-relations-after-introspection).
+
+- **Filtering for `null` and undefined fields**: MongoDB makes a distinction between setting a field to `null` and not setting it at all, which is not present in relational databases. Prisma currently does not express this distinction, which means that you need to be careful when filtering for `null` and undefined fields. For more information, see [How to filter for `null` and undefined fields](#how-to-filter-for-null-and-undefined-fields).
+
+- **Using a replica set**: Prisma uses [MongoDB transactions](https://www.mongodb.com/docs/manual/core/transactions/) internally to avoid partial writes on nested queries. When using transactions, MongoDB requires you to have a [replica set](https://www.mongodb.com/docs/manual/replication/) configured — this is a group of MongoDB processes that maintain the same data set, providing redundancy and high availability. If you use MongoDB's [Atlas](https://www.mongodb.com/atlas/database) hosting service, the replica set is configured for you, but if you are running MongoDB locally you will need to set this up yourself. For more information, see MongoDB's [guide to deploying a replica set](https://www.mongodb.com/docs/manual/tutorial/deploy-replica-set/).
+
+## How to implement Prisma with MongoDB
+
+This section provides instructions for how to carry out tasks that require steps specific to MongoDB.
+
+### How to migrate old data to new schemas
+
+Migrating your database over time is an important part of the development cycle. During development, you will need to update your schema file (for example, toads new fields), then update the data in your your development environment’s database, and eventually push both the updated schema and the new data to the production database.
```

Comment:
I assume frogs have nothing to do with MongoDB, even if they're generally green. `toads` -> `to add`.

Reply:
```suggestion
Migrating your database over time is an important part of the development cycle. During development, you will need to update your schema file (for example, to add new fields), then update the data in your your development environment’s database, and eventually push both the updated schema and the new data to the production database.
```